addons/popochiu/engine/objects/region/popochiu_region.gd

```diff
@@ -121,48 +126,71 @@ func _check_scaling(
 	area_rid: RID, area: Area2D, area_shape_index: int, local_shape_index: int, entered: bool
 ):
 	if not (
-		area is PopochiuCharacter 
-		and area.get_node_or_null("ScalingPolygon") 
-		and area_shape_index == area.get_node("ScalingPolygon").get_index()
+		area is PopochiuCharacter
+		and area.get("scaling_polygon")
+		and area_shape_index == area.get("scaling_polygon").get_index()
 	):
 		return
 	
-	if not entered:
-		_clear_scaling_region(area)
-		return
+	var character: PopochiuCharacter = area
+	# Track character entry/exit across all shapes
+	if entered:
+		_active_characters[character.script_name] = area
+	else:
+		# Only remove if character is not in any other shape
+		var overlapping := get_overlapping_areas()
+		var still_inside := false
+		for overlapping_area in overlapping:
+			if overlapping_area == character:
+				still_inside = true
+				break
 		
-	if scaling:
-		_update_scaling_region(area)
-		area.update_scale()
+		if not still_inside:
+			_active_characters.erase(character.script_name)
+			_clear_scaling_region(character)
+			return
+	
+	if scaling and _active_characters.has(character.script_name):
+		_update_scaling_region(character)
+		character.update_scale()
 
 
 func _update_scaling_region(chr: PopochiuCharacter) -> void:
-	var polygon_y_array = []
-	for x in get_node("InteractionPolygon").get_polygon():
+	var polygon_y_array := []
+	for x: Vector2 in interaction_polygon_node.get_polygon():
 		polygon_y_array.append(x.y)
-
-	chr.on_scaling_region= {
-		"region_description": self.description,
-		"scale_top": self.scale_top, 
-		"scale_bottom": self.scale_bottom,
-		"scale_max": [self.scale_top, self.scale_bottom].max(),
-		"scale_min": [self.scale_top, self.scale_bottom].min(),
-		"polygon_top_y": (
-			polygon_y_array.min()+self.position.y+get_node("InteractionPolygon").position.y 
-			if self.position 
-			else ""
-			),
-		"polygon_bottom_y": (
-			polygon_y_array.max()+self.position.y+get_node("InteractionPolygon").position.y 
-			if self.position 
-			else ""),
-
-		}
+	
+	# Get global positions for more accurate calculations 
+	var global_top: float = (
+		polygon_y_array.min() + global_position.y + interaction_polygon_node.position.y
+	)
+	var global_bottom: float = (
+		polygon_y_array.max() + global_position.y + interaction_polygon_node.position.y
+	)
+	_last_char_pos = chr.global_position
+
+	var region_height := global_bottom - global_top
+	var position_ratio := (chr.global_position.y - global_top) / region_height
+	position_ratio = clamp(position_ratio, 0.0, 1.0)
```

Comment:
Unsure about where this is coming from and how it is used, because I can't see it in the original diff.
Just check this is not a leftover. Maybe I'll move it down in the code so you focus on calculating the vars that will populate the dictionary and than this one?

Reply:
It is used to calculate the target scale for the character. Is part of the fixes needed for the update to Godot 4.4.